mk/docker.mk

```diff
@@ -44,8 +44,8 @@ image/base-root: ## Dev: Rebuild `kuma-base-root` Docker image
 	docker build -t $(KUMA_BASE_ROOT_IMAGE) ${DOCKER_BUILD_ARGS} --build-arg ARCH=${GOARCH} --platform=linux/${GOARCH} -f $(TOOLS_DIR)/releases/dockerfiles/Dockerfile.base-root .
 
 .PHONY: image/envoy
-image/envoy: build/artifacts-linux-${GOARCH}/envoy/envoy ## Dev: Rebuild `envoy` Docker image
-	docker build -t $(KUMA_ENVOY_IMAGE) ${DOCKER_BUILD_ARGS} --build-arg ARCH=${GOARCH} --platform=linux/${GOARCH} --build-arg ENVOY_VERSION=${ENVOY_VERSION} -f $(TOOLS_DIR)/releases/dockerfiles/Dockerfile.envoy .
+image/envoy: build/envoy/linux-${GOARCH} ## Dev: Rebuild `envoy` Docker image
```

Comment:
what is the reason of diverging this path from what we have in master?
I see that on master we store it in 
```
# create targets to build binaries for each OS/ARCH combination
# $(1) - GOOS to build for
# $(2) - GOARCH to build for
build/artifacts-$(1)-$(2)/envoy
```

Reply:
My reasoning was that all the build steps in the current `build.mk` file have one pattern `build/<binary>/$GOOS` I wanted to make them the same. I didn't want to backport all the changes from the master branch, just the one required to backport.